pkg/api/mlflow/dao/repositories/helpers.go

```diff
@@ -0,0 +1,38 @@
+package repositories
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+)
+
+// makeSqlPlaceholders collects a string of (?,?,?), (?,?,?), etc
+func makeSqlPlaceholders(numberInEachSet, numberOfSets int) string {
+	placeholderArray := make([]string, numberInEachSet)
+	for i := 0; i < numberInEachSet; i++ {
+		placeholderArray[i] = "?"
+	}
+	setsArray := make([]string, numberOfSets)
+	for i := 0; i < numberOfSets; i++ {
+		setsArray[i] = fmt.Sprintf("(%s)", strings.Join(placeholderArray, ","))
+	}
+	return strings.Join(setsArray, ",")
+}
+
+// makeParamConflictPlaceholdersAndValues provides sql placeholders and concatenates
+// Key, Value, RunID from each input Param for use in sql values replacement
+func makeParamConflictPlaceholdersAndValues(params []models.Param) (string, []interface{}) {
+	// make place holders of 3 fields for each param
+	placeholders := makeSqlPlaceholders(3, len(params))
+	// values array is params * 3 in length since using 3 fields from each
+	valuesArray := make([]interface{}, len(params)*3)
+	index := 0
+	for _, param := range params {
+		valuesArray[index] = param.Key
+		valuesArray[index+1] = param.Value
+		valuesArray[index+2] = param.RunID
+		index = index + 3
+	}
+	return placeholders, valuesArray
+}
```

Comment:
we can move of course with this solution but my proposal was to have a function based on generics, where you can get the number of properties through the reflection and build `placeholders` for any slice. @jgiannuzzi what do you think?

Reply:
agreed, I would have liked to see something more generic — however this is good enough for this PR